raiden/transfer/architecture.py

```diff
@@ -142,7 +142,28 @@ def __init__(self, sender):
         self.sender = sender
 
     def __eq__(self, other):
-        return self.sender == other.sender
+        return (
+            isinstance(other, AuthenticatedSenderStateChange) and
```

Comment:
We always need to check the type of `other` prior to accessing the attributes, otherwise an `AttributeError` can be raised.